PXWeb/Middleware/AdminProtectionKeyMiddleware.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using System.Threading.Tasks;
+using PxWeb.Config.Api2;
+using System.Collections.Generic;
+using System.Net;
+using System.Linq;
+
+namespace PxWeb.Middleware
+{
+    // You may need to install the Microsoft.AspNetCore.Http.Abstractions package into your project
+    public class AdminProtectionKeyMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly IAdminProtectionConfigurationService _adminProtectionConfigurationService;
+
+        public AdminProtectionKeyMiddleware(RequestDelegate next, IAdminProtectionConfigurationService adminProtectionConfigurationService)
+        {
+            _next = next;
+            _adminProtectionConfigurationService = adminProtectionConfigurationService;
+        }
+
+        public async Task Invoke(HttpContext httpContext)
+        {
+            AdminProtectionConfigurationOptions options = _adminProtectionConfigurationService.GetConfiguration();
```

Comment:
Read options in constructor so we do not need to do it in every call

Reply:
Should be fixed in latest commit

Reply:
Looks good